### ISSUE TITLE
MapChart: draw chart in a clusteredBullet?

### ISSUE THREAD

**Issue by pvougny** [amcharts/amcharts5#1299]
**Question**

My chart is `am5map.MapChart` which has two series: `am5map.MapPolygonSeries`, `am5map.ClusteredPointSeries`.

In the `ClusteredPointSeries`, I can push some `bullet` with a sprite that is a container with a `am5percent.PieSeries`.
I'd like to draw pie charts as well in the `clusteredBullet` using the same mechanism as in the simple bullet. It seems to be not supported. I can push some `am5.Circle` and a `am5.Label` like in the tutorial. Can I push a `am5percent.PieSeries`?

Thank you for your feedback.

**Environment**

 - amCharts version 5.5.6
 - Angular 15

**Screenshot with pie charts (bullets) and labelled circles (clusters)**

<img width="539" alt="mapchart-bullets-clusters" src="https://github.com/amcharts/amcharts5/assets/34480075/53e9b3b6-e14f-4852-9ea1-6446c78480f7">

**Comment by martynasma** [comment]
> I can push some am5.Circle and a `am5.Label` like in the tutorial. Can I push a `am5percent.PieSeries`?

Well, you can push `am5percent.PieChart` as a bullet. `PieSeries` will not work standalone without a `PieChart`.

**Comment by pvougny** [comment]
That's right. I wanted to say pushing a `am5percent.PieChart`. Here is a simple snippet that summarizes my trial:

```typescript
let root = am5.Root.new('chartdiv');
let chart = root.container.children.push(am5map.MapChart.new(root, {});
let pointSeries = chart.series.push(
  am5map.ClusteredPointSeries.new(root, {})
);
```

Clusters work with circle and label:

```typescript
pointSeries.set('clusteredBullet', (root, series, dataItem) => {
  let container = am5.Container.new(root, {});

  container.children.push(am5.Circle.new(root, { ... }));
  container.children.push(am5.Label.new(root, { ... }));

  return am5.Bullet.new(root, {
    sprite: container
  });
});
```

Clusters don't support PieChart:

```typescript
pointSeries.set('clusteredBullet', (root, series, dataItem) => {
  let piechart = root.container.children.push(
    am5percent.PieChart.new(root, { ... })
  );

  let pieseries = piechart.series.push(
    am5percent.PieSeries.new(root, { ... })
  );

  // at this stage, dataItem.dataContext is undefined unlike in bullets
  pieseries.data.setAll(dataItem.dataContext?.data);

  return am5.Bullet.new(root, {
    sprite: piechart,
  });
});
```

`dataItem.dataContext` is empty then I can't summarize all single points. Apart from that, even with hard data, the PieChart does not display.

**Comment by martynasma** [comment]
Would you be able to post your whole chart and data on CodePen/jsFiddle so that we can test?

**Comment by pvougny** [comment]
https://codepen.io/pvougny/pen/gOEGLjd?editors=0010

FIXME is at line 517, where the pieChart is not considered for rendering.

**Comment by pvougny** [comment]
Many thanks for reading me and considering this need!

**Comment by martynasma** [comment]
Fixed in [5.8.0](https://github.com/amcharts/amcharts5/releases/tag/5.8.0).

## [5.8.0] - 2024-02-01

### Added
- New container type `ZoomableContainer` added, which allows adding zoom capabilities to virtually any chart. [More info](https://www.amcharts.com/docs/v5/concepts/common-elements/containers/#Zoomable_container).
- New class `ZoomTools` added. Can be used to quickly add zoom support for elements compatible with `IZoomable` interface, e.g. `ZoomableContainer`.
- New read-only property of `DrawingControl`: `drawingSeries`. Contains an object where key is a drawing tool name and values are array with references to actual drawing series.

### Changed
- Default value for `strokeOpacity` for the grid line of a `StockChart` changed from `0` to `0.4` so that it would differ from regular grid lines.

### Fixed
- Resizing stock chart's panels after moving them up/down was not working properly.
- `Hierarchy.addChildData` was duplicating node's children which were added before.
- If `topDepth` was set to `1` on `Partition`, initially the chart was showing not all levels of nodes.
- When clicking on particular partitions of a `Partion` chart, zoom animation was not playing.
- `DateRangeSelector` was showing extra day in automatically-calculated end date.
- `DateRangeSelector` was zooming to the start of the selected day, instead of end.
- Tooltip background color passed to `HeatLegend.showValue(value, text, color)` was being ignored.
- `ClusteredPointSeries` was not showing charts if they were added to clustered bullets.
- Fixed issue with `DrawingControl` which was unnecessarily duplicating drawing series with each tool selection / API call.
- Using "1M" button in `PeriodSelector` was not always selecting full month.
- Restoring drawings with `DrawingControl.unserializeDrawings()` will now correctly update indexes of the restored drawings based on current drawing counter.
- In some cases `StockChart` was taking a data item which was just outside of zoom range when calculating aggregate values for its series, affecting percent change values and some other aggregates.
- Legend value was not being updated when series axis' range changed. This caused `legendRangeLabelText` to not being updated (if set) when chart was being zoomed-in/out.
- If a type of `StockChart` series was changed with data grouping enabled, some duplicate candlesticks or ohlc sticks were displayed until first zoom.
- In some rare cases chart would show incorrect last segment of a `LineSeries` when in percent mode.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
